src/simtools/settings.py

```diff
@@ -45,10 +45,31 @@ def load(self, args=None, db_config=None):
             "SIMTOOLS_CORSIKA_PATH",
             args.get("corsika_path", None) if args is not None else None,
         )
-        self._corsika_exe = os.getenv(
-            "SIMTOOLS_CORSIKA_EXECUTABLE",
-            args.get("corsika_executable", "corsika") if args is not None else "corsika",
+        self._corsika_exe = self._get_corsika_exec() if self._corsika_path is not None else None
+
+    def _get_corsika_exec(self):
+        """
+        Get the CORSIKA executable from environment variable or command line argument.
+
+        Build the executable name based on configured interaction models. Fall back to
+        legacy naming (simply "corsika") if models are not specified.
+        """
+        he_model = os.getenv(
```

Comment:
Isn't this reading of the env here contradict the usage of `_fill_from_environmental_variables` in `configurator.py`?

Reply:
We need the os.getenv to allow to have applications with now command line parameters (e.g. `simtools-print-version`).

However, you are right that the call is inverted and I've fixed this. It should be:

```
        he_model = (
            self._args.get("corsika_he_interaction")
            if self._args is not None and "corsika_he_interaction" in self._args
            else os.getenv("SIMTOOLS_CORSIKA_HE_INTERACTION")
        )
```

This allows to configure the HE model through the command line and overwrite the ENV variables. Thanks for the hint!